theme/js/dataset-recommendations.js

```diff
@@ -0,0 +1,67 @@
+const recoUrl = 'https://raw.githubusercontent.com/etalab/datasets_reco/master/reco_json/';
+const maxRecos = 2;
+
+function getDatasetId() {
+    const selector = '#json_ld';
+    const dataset = JSON.parse(document.querySelector(selector).text);
+    return dataset && dataset['@id'];
+}
+
+function addRecos(recos) {
+    const recoContainer = document.getElementById('dataset-recommendations-container');
+    let recoChildContainer, recoChildEmbed;
+    recos.splice(0, maxRecos).forEach((reco) => {
+        recoChildContainer = document.createElement('div');
+        recoChildContainer.classList.add('recommendation');
+    
+        recoChildEmbed = document.createElement('div');
+        recoChildEmbed.setAttribute('data-udata-dataset-id', reco[0]);
+    
+        recoChildContainer.appendChild(recoChildEmbed);
+        recoContainer.appendChild(recoChildContainer);
+    });
+}
+
+function addWidgetScript() {
+    const scriptElm = document.createElement('script');
+    scriptElm.type = 'application/javascript';
+    scriptElm.src = '/static/widgets.js';
+    scriptElm.id = 'udata';
+    scriptElm.onload = loadDatasets;
+    document.body.appendChild(scriptElm);
+}
+
+function loadDatasets() {
+    udataScript.loadDatasets();
+}
+
+function fetchRecos(datasetId) {
+    // XXX we can count on fetch polyfill from udata/front/js/mixin.js, right?
+    fetch(recoUrl + datasetId + '.json').then((response) => {
```

Comment:
Error handling (`catch`) is missing in case of:
- remote file is not reachable
- remote file is unparseable

Reply:
👍

Reply:
I just added a general `catch` with a log, I think it's enough. Not reachable won't break FYI.